napalm_panos/panos.py

```diff
@@ -0,0 +1,326 @@
+                                               'config.')
+            self._import_file(filename)
+
+            if self.connected is False:
+                self._open_ssh()
+
+            cmd = ("load config partial from {0} "
+                  "from-xpath {1} to-xpath {2} mode {3}".format(filename,
+                                                                from_xpath,
+                                                                to_xpath,
+                                                                mode))
+            self.connected.send_config_set([cmd])
+            self.loaded = True
+
+        elif config:
+            if self.loaded is False:
+                if self._save_backup() is False:
+                    raise MergeConfigException('Error while storing backup '
+                                               'config.')
+
+            if self.connected is False:
+                self._open_ssh()
+
+            if isinstance(config, str):
+                config = config.splitlines()
+
+            for command in config:
+                if 'set' not in command:
+                    raise MergeConfigException('Unsupported config format. '
+                                               'You should use "set" commands.'
+                                               )
+
+            self.connected.send_config_set(config)
+            self.loaded = True
+
+        else:
+            raise MergeConfigException('You must provide either an XML file '
+                                       'or a set-format string')
+
+    def compare_config(self):
+        if self.connected is False:
+            self._open_ssh()
+
+        output = self.connected.send_command("show config diff")
+        print output
+
+    def _save_backup(self):
+        self.backup_file = 'config_{0}.xml'.format(str(datetime.now().date()).replace(' ', '_'))
+        backup_command = '<save><config><to>{0}</to></config></save>'.format(self.backup_file)
+
+        self.device.op(cmd=backup_command)
+        if self.device.status == 'success':
+            return True
+        else:
+            return False
+
+    def commit_config(self):
+        if self.loaded:
+            self.device.commit(cmd="<commit></commit>")
+
+            if self.device.status == 'success':
+                self.loaded = False
+                self.changed = True
+                time.sleep(15)
+            else:
+                raise ReplaceConfigException('Error while commiting config')
+        else:
+            raise ReplaceConfigException('No config loaded.')
+
+    def discard_config(self):
+        if self.loaded:
+            discard_cmd = '<load><config><from>{0}</from></config></load>'.format(self.backup_file)
+            self.device.op(cmd=discard_cmd)
+
+            if self.device.status == 'success':
+                self.loaded = False
+            else:
+                print "Error while loading last-saved config."
```

Comment:
Please avoid print -- better raise exceptions (https://github.com/napalm-automation/napalm-base/blob/master/napalm_base/exceptions.py). If you need a different one, feel free to open a PR for a new one.